packages/paste-core/components/toast/src/Toaster.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import {useTransition, animated} from '@twilio-paste/animation-library';
+import {Box} from '@twilio-paste/box';
+import {Toast} from './Toast';
+import {ToastPortal} from './ToastPortal';
+import {ToasterProps, ToasterPropTypes} from './types';
+
+export const AnimatedToast = animated(Box);
+
+interface ReturnTriggerState {
+  trigger: HTMLElement | null;
+}
+
+const Toaster: React.FC<ToasterProps> = ({toasts, pop, ...props}) => {
+  const [refMap] = React.useState(() => new WeakMap());
```

Comment:
My only concern here is I can't seem to remove anything from it when I'm popping toasts from the stack. `refMap.delete(item)` "works", air quotes, in that if I step through it and inspect the map before and after the item is deleted. But it doesn't actually see to persist or reflect in the map used in the state

Reply:
I don't quite follow. Are you saying that manually calling "delete()" works, but by deleting the item the `refMap` still logs it as part of the state? The UI seems to respond to the changes correctly, so it could just be a weird React thing? Have you tried forcing garbage collection to see if that cleared it: https://developers.google.com/web/tools/chrome-devtools/memory-problems#visualize_memory_leaks_with_timeline_recordings ?

Reply:
hmmm, yeah it actually seems kinda fine on the memory front.